.github/workflows/tests.yml

```diff
@@ -207,13 +207,12 @@ jobs:
           pytest -v tests/patched/ --cov=axolotl --cov-append --cov-report=xml
           pytest -v tests/cli/ --cov=axolotl --cov-append --cov-report=xml
 
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v5
+      - name: Upload coverage artifacts
+        uses: actions/upload-artifact@v4
         with:
-          token: ${{ secrets.CODECOV_TOKEN }}
-          files: ./coverage.xml
-          flags: unittests,pytorch-${{ matrix.pytorch_version }}
-          fail_ci_if_error: false
+          name: coverage-${{ matrix.pytorch_version }}
+          path: ./coverage.xml
+          retention-days: 1
```

Comment:
Not too clear about GH action, but are these artifacts shared across runs?

Do we need to make sure that the different PR CIs don't download the same artifacts if they're running simultaneously ?